src/Uno.UI/DataBinding/BindingPath.BindingItem.cs

```diff
@@ -198,7 +198,7 @@ private void OnDataContextChanged()
 
 			private void OnPropertyChanged(object? previousValue, object? newValue, bool shouldRaiseValueChanged)
 			{
-				if (_isDataContextChanging && newValue is UnsetValue)
+				if (_isDataContextChanging && newValue == DependencyProperty.UnsetValue)
```

Comment:
Here as well, it's just `ceq`:

![image](https://github.com/unoplatform/uno/assets/31348972/29239f87-9508-4959-88a3-eb23b5f1e8dc)